mubench.pipeline/tests/data/test_build_command.py

```diff
@@ -0,0 +1,270 @@
+        uut = BuildCommand.get("-some_command-")
+
+        assert_is_instance(uut, BuildCommand)
+        assert_equals("-some_command-", uut._name)
+
+    def test_raise_on_multiple_matches(self):
+        duplicate_command = BuildCommandTestImpl.TEST_COMMAND
+        self.test_subclasses = [BuildCommandTestImpl, BuildCommandTestImpl]
+
+        assert_raises(ValueError, BuildCommand.get, duplicate_command)
+
+    @patch("data.build_command.Shell.exec")
+    def test_execute_runs_command(self, shell_mock):
+        uut = BuildCommand.get("-command-")
+
+        uut.execute("-project_dir-", "-dep_dir-", "-cmp_base_path-")
+
+        shell_mock.assert_called_with("-command-",
+                                      cwd="-project_dir-", logger=ANY)
+
+    @patch("data.build_command.Shell.exec")
+    def test_execute_runs_command_with_args(self, shell_mock):
+        command = "-command- -arg- --arg--"
+        uut = BuildCommand.get(command)
+
+        uut.execute("-project_dir-", "-dep_dir-", "-cmp_base_path-")
+
+        shell_mock.assert_called_with(command,
+                                      cwd="-project_dir-", logger=ANY)
+
+    @patch("data.build_command.Shell.exec")
+    def test_execute_copies_dependencies(self, shell_mock):
+        shell_mock.return_value = "-output-"
+        command = "-command-"
+        uut = BuildCommand.get(command)
+        uut._copy_dependencies = MagicMock()
+
+        uut.execute("-project_dir-", "-dep_dir-", "-cmp_base_path-")
+
+        uut._copy_dependencies.assert_called_with("-output-",
+                                                  "-project_dir-",
+                                                  "-dep_dir-",
+                                                  "-cmp_base_path-")
+
+    def test_saves_args(self):
+        uut = BuildCommand.get("-some_command- arg1 arg2")
+
+        assert_equals(["arg1", "arg2"], uut.args)
+
+    def test_quoted_args_are_saved_without_quotes(self):
+        uut = BuildCommand.get("-some_command- 'arg 1' \"arg 2\"")
+
+        assert_equals(["arg 1", "arg 2"], uut.args)
+
+    @patch("data.build_command.Shell.exec")
+    def test_raises_on_error(self, shell_mock):
+        shell_mock.side_effect = CommandFailedError("-command-", "-output-")
+        uut = BuildCommand.get("-some_command-")
+
+        assert_raises(CommandFailedError, uut.execute, "-p-", "-d-", "-bp-")
+
+    @patch("data.build_command.Shell.exec")
+    def test_default_does_not_filter_output_on_error(self, shell_mock):
+        shell_mock.side_effect = CommandFailedError("-command-", "-output-")
+        uut = BuildCommand.get("-some_command-")
+
+        try:
+            uut.execute("-p-", "-d-", "-bp-")
+        except CommandFailedError as e:
+            assert_equals("\n-output-", e.output)
+
+
+@patch("data.build_command.shutil.copy")
+@patch("data.build_command.Shell.exec")
+class TestMavenCommand:
+    def test_compile_with_maven(self, shell_mock, copy_mock):
```

Comment:
Right. I'll do a cleanup now that the commands are mostly set in place. I just wanted to make sure that everything still runs.